src/lib.rs

```diff
@@ -490,13 +495,54 @@ impl Game {
             }
             Message::FinishedTask(FinishedTask { color, index }) => {
                 for player in self.players.iter_mut() {
-                    if player.color == *color {
-                        if let Some(task) = player.tasks.get_mut(*index) {
+                    if player.color == color {
+                        if let Some(task) = player.tasks.get_mut(index) {
                             task.finished = true;
                         }
                     }
                 }
             }
+            Message::Snapshot(Snapshot { bodies, players }) => {
+                // This is a sloppy kind of consensus that isn't guaranteed to converge.
+                // Easiest way to resolve this is to make the server smarter, but that will
+                // involve learning tokio D:
```

Comment:
Is it more complicated because the server currently just relays messages?

Naively, I don't understand why the server cannot hold on to canonical state without the help of another crate.

Reply:
It's more complicated because I'll need to learn how `async/await` and the main async crate, `tokio` work.

Since writing this comment, I've gotten the very rough basics, and I think I'm ready to start hacking on that (though the fact that immediately after I started messing with the server code, the client code started panicking isn't as promising...)